spec/support/shared_examples/api_change_application_funded_plance_endpoint_support.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples "an API change application funded place endpoint" do
+  context "when authorized" do
+    let(:params) { { data: { type: "npq-application-accept", attributes: { funded_place: true } } } }
+
+    it "updates funded place" do
+      expect { api_put(path(application_id), params:) }
+        .to change { application.reload.funded_place }.to be_truthy
+    end
+
+    it "responds with 200 and representation of the application" do
+      api_put(path(application_id), params:)
+
+      expect(response).to be_successful
+      expect(parsed_response.dig("data", "attributes", "funded_place")).to be_truthy
+    end
+
+    context "when funded_place is nil in the params" do
+      let(:params) { { data: { type: "npq-application-accept", attributes: { funded_place: nil } } } }
+
+      it "return 422" do
+        api_put(path(application_id), params:)
+
+        expect(response).to have_http_status(:unprocessable_entity)
+      end
+
+      it "returns error in response" do
+        api_put(path(application_id), params:)
+
+        expect(parsed_response).to be_key("errors")
+        expect(parsed_response.dig("errors", 0, "detail")).to eql("The entered '#/funded_place' is missing from your request. Check details and try again.")
+      end
+    end
+
+    context "with no params" do
+      let(:params) { {} }
+
+      it "return 422" do
+        api_put(path(application_id), params:)
+
+        expect(response).to have_http_status(:unprocessable_entity)
+      end
+
+      it "returns error in response" do
+        api_put(path(application_id), params:)
+
+        expect(parsed_response).to be_key("errors")
+        expect(parsed_response.dig("errors", 0, "detail")).to eql("The entered '#/funded_place' is missing from your request. Check details and try again.")
+      end
+    end
+
+    context "when application is not accepted" do
```

Comment:
Are we over testing the change service here? the validation of the service is already tested in the service class specs so maybe mocking it out and having a single test that covers when there is an error it returns a 422 would suffice?